src/Composer/Command/CompileAssets.php

```diff
@@ -73,10 +80,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $plugin = new Plugin();
             $plugin->activate($composer, $io);
 
-            $noDev = (bool)$input->hasOption('no-dev');
+            $noDev = $input->hasOption('no-dev');
             $env = $input->hasOption('env') ? $input->getOption('env') : null;
 
-            $plugin->runByCommand(is_string($env) ? $env : null, !$noDev);
+            $ignoreLock = '';
+            if ($input->hasOption('ignore-lock')) {
+                $ignored = $input->getOption('ignore-lock');
```

Comment:
@gmazzap The call to `$input->hasOption('ignore-lock')` is always returning true even when I don't set the `--ignore-lock` option and `$input->getOption('ignore-lock')` always return an empty string if I don't pass a value, so just adding the option without a value will not fallback to setting the value to `*` and no asset compilation will be forced.

Reply:
Thanks @cristianobaptista

According to Symfony documentation, `$input->getOption` is supposed to to return the default value when flag is provided without value, and return `null` when the flag is not there at all... but for some reason it's not working like that, maybe it's Composer that changes the behavior.

Switching to the low-level `hasParameterOption` should have solved the issue now.

Could you please test again?

Reply:
@gmazzap All good now after performing once again all tests, thanks!